transport/owr_crypto_utils.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright (c) 2015, Ericsson AB. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice, this
+ * list of conditions and the following disclaimer in the documentation and/or other
+ * materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY
+ * OF SUCH DAMAGE.
+ */
+
+/*/
+\*\ Owr crypto utils
+/*/
+
+/**
+ * 
+ *
+ * Functions to get certificate, fingerprint and private key.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "owr_crypto_utils.h"
+#include "owr_private.h"
+#include "owr_utils.h"
+
+
+#ifdef OWR_STATIC
+#include <stdlib.h>
+#endif
+
+#ifdef __APPLE__
+#include <TargetConditionals.h>
+#endif
+
+#ifdef __ANDROID__
+#include <android/log.h>
+#endif
+
+#include <openssl/ssl.h>
+#include <openssl/evp.h>
+#include <openssl/rsa.h>
+
```

Comment:
Please add a debug category for owrcrypto, else GST_DEBUG() and such will use the 'default' (literally) category.

Add this here:
```
GST_DEBUG_CATEGORY_EXTERN(_owrcrypto_debug);
#define GST_CAT_DEFAULT _owrcrypto_debug
```

The following at https://github.com/EricssonResearch/openwebrtc/blob/master/owr/owr.c#L72 :
```
GST_DEBUG_CATEGORY(_owrcrypto_debug);
```

And the following here https://github.com/EricssonResearch/openwebrtc/blob/master/owr/owr.c#L225 :
```
    GST_DEBUG_CATEGORY_INIT(_owrcrypto_debug, "owrcrypto", 0,
        "OpenWebRTC Cryptography");
```